File: backend/src/main/java/com/smartestate/entity/Repair.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.smartestate.constants.RepairConstants;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("repairs")
public class Repair {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private String type;
    private String images;
    private String status = RepairConstants.PENDING;
    private Long handlerId;
    private Integer rating;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @TableField(exist = false)
    private User user;
    @TableField(exist = false)
    private User handler;
}

File: frontend/src/types/repair.ts
import type { RepairStatus } from '../constants/repair';
import type { User } from './user';

export interface Repair {
  id: number;
  userId: number;
  title: string;
  description: string;
  type: 'water_power' | 'furniture' | 'public_facility' | 'other';
  images?: string;
  status: RepairStatus;
  handlerId?: number;
  rating?: number;
  createdAt: string;
  updatedAt: string;
  user?: User;
  handler?: User;
}

export interface RepairPayload {
  title: string;
  description: string;
  type: Repair['type'];
}
